nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -519,16 +525,41 @@ private Description checkForReadBeforeInit(ExpressionTree tree, VisitorState sta
       // also suppress checking read before init, as we may not find explicit initialization
       return Description.NO_MATCH;
     }
-    return checkPossibleUninitFieldRead(
-        tree, state, symbol, path, enclosingBlockPath, enclosingClassPath);
+    return checkPossibleUninitFieldRead(tree, state, symbol, path, enclosingBlockPath);
   }
 
-  private boolean relevantInitializerMethodOrBlock(Tree methodLambdaOrBlock, VisitorState state) {
+  private Symbol.ClassSymbol enclosingClassSymbol(TreePath enclosingBlockPath) {
+    Tree leaf = enclosingBlockPath.getLeaf();
+    if (leaf instanceof BlockTree) {
+      // parent must be a ClassTree
+      Tree parent = enclosingBlockPath.getParentPath().getLeaf();
+      return ASTHelpers.getSymbol((ClassTree) parent);
```

Comment:
Just out of curiosity, this is an optimization over the else branch, right? `ASTHelpers.enclosingClass(ASTHelpers.getSymbol(leaf))` will always get the enclosing class and is only more expensive?

Reply:
No the issue is that the `else` branch uses `ASTHelpers.getSymbol(leaf)` which i don't think will work for a `BlockTree`